IntegrationTests/LayoutEventsTest.js

```diff
@@ -36,56 +35,80 @@ type State = {
   containerStyle?: ViewStyleProp,
 };
 
-const LayoutEventsTest = createReactClass({
-  displayName: 'LayoutEventsTest',
-  getInitialState(): State {
-    return {
-      didAnimation: false,
-    };
-  },
-  animateViewLayout: function() {
+class LayoutEventsTest extends React.Component<$FlowFixMeProps, State> {
+  _view: ?View;
+  _img: ?Image;
+  _txt: ?Text;
+  _isMounted: boolean = true;
+
+  state: State = {
+    didAnimation: false,
+  };
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  animateViewLayout = () => {
     debug('animateViewLayout invoked');
     LayoutAnimation.configureNext(LayoutAnimation.Presets.spring, () => {
       debug('animateViewLayout done');
       this.checkLayout(this.addWrapText);
     });
     this.setState({viewStyle: {margin: 60}});
-  },
-  addWrapText: function() {
+  };
+
+  addWrapText = () => {
     debug('addWrapText invoked');
     this.setState(
       {extraText: '  And a bunch more text to wrap around a few lines.'},
       () => this.checkLayout(this.changeContainer),
     );
-  },
-  changeContainer: function() {
+  };
+
+  changeContainer = () => {
     debug('changeContainer invoked');
     this.setState({containerStyle: {width: 280}}, () =>
       this.checkLayout(TestModule.markTestCompleted),
     );
-  },
-  checkLayout: function(next?: ?Function) {
-    if (!this.isMounted()) {
+  };
+
+  checkLayout = (next?: ?Function) => {
+    if (!this._isMounted) {
```

Comment:
Hmm... It seems like we were using the `isMounted()` check to ensure that `this.refs.{view,txt,img}`were all non-null. I wonder if we should replace this check with: `this._view == null || this._txt == null || this._img == null`.

```JavaScript
  if (this._view == null || this._txt == null || this._img == null) {
    return;
  }
  
  this._view.measure((x, y, width, height) => {
    this.compare('view', {x, y, width, height}, this.state.viewLayout);
    if (typeof next === 'function') {
      next();
    } else if (!this.state.didAnimation) {
      // Trigger first state change after onLayout fires
      this.animateViewLayout();
      this.state.didAnimation = true;
    }
  });
  this._txt.measure((x, y, width, height) => {
    this.compare('txt', {x, y, width, height}, this.state.textLayout);
  });
  this._img.measure((x, y, width, height) => {
    this.compare('img', {x, y, width, height}, this.state.imageLayout);
  });  

```

If we do this, we could get rid of `isMounted` completely.

Reply:
Since it's only used here I think we can get rid of isMounted, I'll update it